queries/query.go

```diff
@@ -95,6 +95,58 @@ type with struct {
 	args   []interface{}
 }
 
+type orderBy struct {
+	clause string
+	args   []interface{}
+}
+
+type modifiers interface {
```

Comment:
I think this interface is overkill. If you look at the way it's being used it's only saving us from sending 2 arguments into the `writeParameterizedModifiers` function (as clauses() is duplicate code in all implementations). Unless there's further reasoning I think these types should be removed and simply pass keyword/delim into the function.

Reply:
I've dropped the interface.

What I've replaced it with might also be a little weird.  Now there is a `orderByClauses` helper and a `havingClauses` helper in `query_builders.go` and a one-off struct called `clause`, a slice of which is passed as the third argument.  This was my attempt to avoid duplicating most of the code in `writeParameterizedModifiers`.  But you might have in mind something else.